src/main/java/draylar/identity/cca/IdentityComponent.java

```diff
@@ -101,9 +99,10 @@ else if (type.isPresent()) {
                     ((DimensionsRefresher) player).refresh();
 
                     // refresh flight abilities
+                    // TODO remove when compatibility with old saves is no longer needed
                     if(!player.world.isClient) {
                         if (Identity.hasFlyingPermissions((ServerPlayerEntity) player)) {
-                            player.abilities.allowFlying = true;
+                            Identity.ABILITY_SOURCE.grantTo(player, VanillaAbilities.ALLOW_FLYING);
                         }
                     }
```

Comment:
This code is no longer needed, but removing it right now will result in morphed players losing their flight until they morph again - hence the todo.